mantidproject/mantid
# Fix "Plot surface/contour from group"

Refactor the code for plotting surface/contour plots from a WorkspaceGroup to use `HistogramData`.
This ensures that it now works and does not throw an exception for histogram workspaces.

To avoid making a 3D plot of a ragged workspace, the input workspaces are required to all have the same X data length.

**To test:**

Use the following script:

``` python
def create_sample_group(n_items, histo, name):
    to_group = list()
    xData = range(200, 211, 1) if histo else range(200, 210, 1)
    for i in range(n_items):
        ws_name = name + str(i)
        to_group.append(ws_name)
        CreateWorkspace(DataX=xData, DataY=[i+1]*20, NSpec=2, OutputWorkspace=ws_name)
        AddSampleLog(Workspace=ws_name, LogName="MyLog", LogText=str((i+1)*5), LogType='Number')
    GroupWorkspaces(InputWorkspaces=','.join(to_group), OutputWorkspace="gp_"+name)

create_sample_group(n_items=10, histo=True, name="histo")
create_sample_group(n_items=10, histo=False, name="points")
```

Right-click on each group and choose "Plot surface from group" or "Plot contour from group". Check that you do not see the exception in the issue when making the plot.

If you try this with other workspace groups, note that all workspaces in the group must have the same X data length (or an error message will be displayed).

No tests as this is in MantidPlot.

Fixes #17896.

[Release notes](https://github.com/mantidproject/mantid/blob/64e93cf6d9a04dc34bc8e2ae9bb2a280e26695e5/docs/source/release/v3.9.0/ui.rst#bugs-resolved) have been updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
